concourse/pipelines/artifact-releaser-test.jsonnet

```diff
@@ -0,0 +1,616 @@
+  // of the publish workflows, any environment that is not 'testing' or 'prod' will make the publish
+  // project the running project, in this case 'artifact-releaser-autopush'.
+  env:: 'autopush',
+
+  wf_dir:: error 'must set wf_dir in arle_publish_images_autopush',
+  workflow:: '%s/%s.publish.json' % [self.workflow_dir, underscore(self.image)],
+
+  gcs_dir:: error 'must set gcs_dir in arle-publish-images-autopush',
+  gcs_bucket:: common.prod_bucket,
+  gcs:: 'gs://%s/%s' % [self.gcs_bucket, self.gcs_dir],
+
+  passed:: 'build-' + tl.image,
+
+  name: 'arle-publish-images-autopush-%s' % tl.image,
+  plan: [
+    { get: 'guest-test-infra' },
+    {
+      get: '%s-gcs' % tl.image,
+      trigger: true,
+      params: { skip_download: true },
+    },
+    { task: 'generate-timestamp', file: 'guest-test-infra/concourse/tasks/generate-timestamp.yaml' },
+    { load_var: 'start-timestamp-ms', file: 'timestamp/timestamp-ms' },
+    { load_var: 'source-version', file: '%s-gcs/version' % tl.image },
+    { task: 'generate-version', file: 'guest-test-infra/concourse/tasks/generate-version.yaml' },
+    { load_var: 'publish-version', file: 'publish-version/version' },
+    {
+      task: 'publish-autopush-%s' % tl.image,
+      config: arle.gcepublishtask {
+        source_gcs_path: tl.gcs,
+        source_version: 'v((.:source-version))',
+        publish_version: '((.:publish-version))',
+        wf: tl.workflow,
+        environment: tl.env,
+      },
+    },
+  ],
+  on_success: {
+    task: 'publish-success-metric',
+    config: common.publishresulttask {
+      pipeline: 'artifact-releaser-test',
+      job: 'arle-publish-image-autopush-%s' % tl.image,
+      result_state: 'success',
+      start_timestamp: '((.:start-timestamp-ms))',
+    },
+  },
+  on_failure: {
+    task: 'publish-failure-metric',
+    config: common.publishresulttask {
+      pipeline: 'artifact-releaser-test',
+      job: 'arle-publish-image-autopush-%s' % tl.image,
+      result_state: 'failure',
+      start_timestamp: '((.:start-timestamp-ms))',
+    },
+  },
+};
+
+// Image group definition.
+local imggroup = {
+  local tl = self,
+
+  images:: error 'must set images in imggroup',
+
+  name: error 'must set name in imggroup',
+  jobs: [
+    'arle-publish-images-autopush-' + image
+    for image in tl.images
+  ],
+};
+
+// Package group definition
+local pkggroup = {
+  local tl = self,
+
+  packages:: error 'must set packages in pkggroup',
+  builds:: error 'must set builds in pkggroup',
+
+  name: error 'must set name in pkggroup',
+  jobs: [
+    'upload-arle-autopush-staging-%s' % package
+    for package in tl.packages
+  ],
+};
+
+{
+  // Image names.
+  local almalinux = ['almalinux-8', 'almalinux-9'],
+  local debian = ['debian-10', 'debian-11', 'debian-11-arm64', 'debian-12', 'debian-12-arm64'],
+  local centos = ['centos-7', 'centos-stream-8', 'centos-stream-9'],
+  local rhel = [
+    'rhel-7',
```

Comment:
Side Note: these are some kinds of repetition I'd love if we could stop propagating and start adding a more generic set of configurations and abstraction that make sense in a more global sense, `rhel` or enterprise linux distribution names are stable and equal across all contexts.

Reply:
Acknowledged.